File: backend/app/presentation/api/schemas/requests/contraction.py
from datetime import datetime

from pydantic import BaseModel


class StartContractionRequest(BaseModel):
    intensity: int | None = None
    start_time: datetime | None = None
    notes: str | None = None


class EndContractionRequest(BaseModel):
    intensity: int
    end_time: datetime | None = None
    notes: str | None = None


class UpdateContractionRequest(BaseModel):
    contraction_id: str
    start_time: datetime | None = None
    end_time: datetime | None = None
    intensity: int | None = None
    notes: str | None = None

File: apps/labour/worker/src/durable_object/write_side/domain/commands/contraction.rs
use chrono::{DateTime, Utc};
use serde::{Deserialize, Serialize};
use uuid::Uuid;

#[derive(Debug, Clone, Deserialize, Serialize)]
pub struct StartContraction {
    pub labour_id: Uuid,
    pub contraction_id: Uuid,
    pub start_time: DateTime<Utc>,
}

#[derive(Debug, Clone, Deserialize, Serialize)]
pub struct EndContraction {
    pub labour_id: Uuid,
    pub contraction_id: Uuid,
    pub end_time: DateTime<Utc>,
    pub intensity: u8,
}

#[derive(Debug, Clone, Deserialize, Serialize)]
pub struct UpdateContraction {
    pub labour_id: Uuid,
    pub contraction_id: Uuid,
    pub start_time: Option<DateTime<Utc>>,
    pub end_time: Option<DateTime<Utc>>,
    pub intensity: Option<u8>,
}

#[derive(Debug, Clone, Deserialize, Serialize)]
pub struct DeleteContraction {
    pub labour_id: Uuid,
    pub contraction_id: Uuid,
}
